src/gluonts/dataset/parallelized_loader.py

```diff
@@ -84,16 +85,77 @@ def reduce_ndarray(data):
 ForkingPickler.register(nd.NDArray, reduce_ndarray)
 
 
+def _is_stackable(
+    arrays: List[Union[np.ndarray, mx.nd.NDArray, Any]],
+) -> bool:
+    """
+    Check if elements are scalars, have too few dimensions, or their
+    time axes have equal length; i.e. they are directly `stack` able.
+    """
+    if isinstance(arrays[0], (mx.nd.NDArray, np.ndarray)):
+        s = set(arr.shape[0] for arr in arrays)
+        return len(s) <= 1 and arrays[0].shape[0] != 0
+    return True
+
+
+def _pad_arrays(
+    data: List[Union[np.ndarray, mx.nd.NDArray]], pad_axis: int = 0,
+) -> List[Union[np.ndarray, mx.nd.NDArray]]:
+    assert isinstance(data[0], (np.ndarray, mx.nd.NDArray))
+    is_mx = isinstance(data[0], mx.nd.NDArray)
+
+    # MxNet causes a segfault when persisting 0-length arrays. As such,
+    # we add a dummy pad of length 1 to 0-length dims.
+    max_len = max(1, functools.reduce(max, (x.shape[pad_axis] for x in data)))
+    padded_data = []
+
+    for x in data:
+        # MxNet lacks the functionality to pad n-D arrays consistently.
+        # We fall back to numpy if x is an mx.nd.NDArray.
```

Comment:
So this could mean a lot of overhead, we should be extra sure that there is no way of padding MXNet arrays properly.